### PR TITLE
Avoid creating unnecessary strings in adjustExtension

### PR DESCRIPTION
**Pull request by odersky**

### CODE REVIEW

**Comment by odersky** [comment]
test performance please

**Comment by odersky** [comment]
That one gives a clear win for dotc. For stdlib we see the same pattern as in the optimize-megaphase commit: go back up to the norm after a very low number for optimize-current. But it seems optimize-current is part of these two as well since it is in master. Which leaves as only explanation for me that we got an outlier for stdlib when we tested optimize-current.

File: compiler/src/dotty/tools/dotc/typer/Typer.scala [hunk 1]
```diff
@@ -216,7 +216,10 @@ class Typer extends Namer
         val termName = name.toTermName
 
         def adjustExtension(name: Name) =
-          if required.is(ExtensionMethod) then name.toExtensionName else name
+          if required.is(ExtensionMethod) && termName.endsWith(name.lastPart)
+               // pre-check to avoid forming a new string; form extension only if it has a chance of matching `termName`
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/typer/Typer.scala:220]: Will `termName.endsWith(name.lastPart)` be always true, given that `termName = name.toTermName`?

**Review by liufengyun — APPROVED** [review]
Otherwise, LGTM

**Pull request merged**